draft-ietf-ppm-dap.md

```diff
@@ -2653,59 +2638,57 @@ running aggregation jobs ({{aggregate-flow}}) until the Collector initiates a
 collection job. This is because, in general, the aggregation parameter is not
 known until this point. In certain situations it is possible to predict the
 aggregation parameter in advance. For example, for Prio3 the only valid
-aggregation parameter is the empty string. For these reasons, the collection job
-is handled asynchronously. If aggregation is performed eagerly, the Leader MUST
-validate that the aggregation parameter received in the `CollectionJobReq`
-matches the aggregation parameter used in aggregations.
+aggregation parameter is the empty string.
 
 Upon receipt of a `CollectionJobReq`, the Leader begins by checking that it
-recognizes the task ID in the request path. If not, it MUST abort with error
-`unrecognizedTask`.
+recognizes the task ID. If not, it MUST abort with error `unrecognizedTask`.
+
+Next, the Leader checks that the indicated batch mode matches the task's batch
+mode. If not, then the Leader MUST abort with error `invalidMessage`.
 
 Next, the Leader checks that the aggregation parameter is valid as described in
-{{agg-param-validation}}. If the aggregation parameter is invalid, then the
-Leader MUST abort with error `invalidAggregationParameter`.
+{{agg-param-validation}}. If not, then the Leader MUST abort with error
+`invalidAggregationParameter`.
+
+If aggregation is performed eagerly, the Leader MUST validate that the
+aggregation parameter received in the `CollectionJobReq` matches the aggregation
+parameter used in aggregations.
```

Comment:
```suggestion
If aggregation is performed eagerly, the Leader MUST validate that the
aggregation parameter received in the `CollectionJobReq` matches the aggregation
parameter used in each aggregation job pertaining to the batch.
```

Also, what causes this check to fail? Would this be a internal error on the part of the Leader?

Reply:
It's hard to say. If aggregation was performed eagerly, then either:

- the aggregation parameter for the VDAF is the unit type (`()` in Rust), like for Prio3
- the VDAF does have a non-unit aggregation parameter, but for some reason (special knowledge at the application layer above DAP) its value is known to the Leader such that it could perform eager aggregation.

In the first case, it's not possible for the aggregation parameter values to disagree because there's no aggregation parameter values.

In the second case, I guess it would make sense for the Leader to check that the aggregation parameter it gets from the Collector matches what it guessed when it ran the aggregation jobs.

That's me reaching for a reason to have this check, though.

Reply:
I think validation makes sense, but let's say waht happens when the check fails. I.e., instead of "MUST validate" write "if validation fails, then MUST abort with error BLAH".